src/components/tools/Stats/StatOptions.svelte

```diff
@@ -39,17 +41,56 @@
     }
   }
 
+  /**
+   * When user types in the year, the combobox dropdown menu is filtered to show only
+   * values that complete the characters typed in the combobox
+   **/
+  function shouldFilterItem(item, value) {
+    if (!value) return true;
+    return item.text.includes(value);
+  }
+
+  /**
+   * When user presses the Enter or Tab key in combobox after typing in a year, update
+   * the combobox selected index if the year matches an item in the combobox dropdown menu
+   **/
+  function updateIndex(e) {
+    const { key, keyCode, target } = e;
+    if (["Enter", "Tab"].includes(key) || [13, 9].includes(keyCode)) {
+      const { value, id } = target;
+      switch (id) {
+        case "years-select-start":
+          startYear_selectedIndex = items.findIndex((d) => d.text === value);
```

Comment:
I think part of the bug I'm seeing has to do with how `startYear_selectedIndex` is being set here.

Reply:
That's really strange. So even after you find the correct index using the highlighted class name it still is overridden?

Reply:
Yes. Looking at the code, it look like the component also tracks highlighted id internally. `shouldFilterItem` runs after each character is entered and at this point there are 2 lists for the combobox -the  original with all values and a temporary filtered list with matching values. The index of the value highlighted (e.g. 2032) might be 26 in original list but 2 in the filtered list. So even if I set the `selectedIndex` to 26 (corresponding to 2032), some lifecycle function in the component probably sets it to 2 (corresponding to 2008).